hangman.py

```diff
@@ -0,0 +1,129 @@
+import random
+from words import words
+
+def hangman_pick_word():
+    """
+    Randomly picks a theme from the words.py file and converts to upper case.
+    """
+    theme = random.choice(list(words.keys()))
+    word = random.choice(words.get(theme)).upper()
+    return theme, word
+
+def hangman_generate_hint(theme, word):
+    """
+    Generates a hint based on the category randomly chosen
+    """
+    print("It\'s a {0} with {1} letters!".format(theme, len(word)))
+
+
+def hangman_print_dashes(word):
+    """
+    Generates a standard hint and the number of dashes based on the word
+    previously selected 
+    """
+    hangman_clear_screen()
+    no_of_dashes = len(word)
+    print_dashes = "_" * no_of_dashes 
+    print("Word: {0} ".format(" ".join(print_dashes)))
+    return print_dashes
+
+
+def hangman_get_letter():
+    """
+    Requests user to input a letter and converts it to upper case
+    """
+    user_guess = input("Please guess a letter: ").upper()
+    hangman_clear_screen()
+    return user_guess
+
+
+def hangman_clear_screen():
+    """
+    Clears screen based on the range specifed.
+    """
+    for i in range(1):
+        print("\n")
+
+
+def hangman_win_lose(word, guessed):
+    """
+    Checks if the player won or lost and prints the information
+    """
+    if guessed:
+        print("Congratulations! You won the game!")
+    else:
+        print("Sorry you ran out of lives! The correct answer is {0} ".format(word))
+    return 
+
+
+def hangman_algorithm(theme, word):
+    """
+    Sets user attemps and initial state of the game. Obtains user guesses and
+    checks for validity. If user guess is correct, displays the appropriate
+    letters corresponding to the guess. If the user guess is invalid, decreases
+    attempts and informs user. If user guessed a previously guessed letter, no
+    penealty for the user.
+    """
+    user_guess_list = set()
+    attempts = 9
+    guessed = False
+    valid_indices = []
+
+    print_dashes = hangman_print_dashes(word)
+    hangman_generate_hint(theme, word)
+
+    print_dashes_list = list(print_dashes)
+    word_list = list(word)
+
+    while not guessed and attempts > 0:
+        print("Remaining Lives: {0} ".format(attempts))
+        print("Letters Guessed: {0} ".format(user_guess_list))
+        user_guess = hangman_get_letter()
+
+        if (len(user_guess) == 1 and user_guess.isalpha()):
+            if user_guess in user_guess_list:
+                print("You've already guessed this letter! Try again!")
+                print("Word: {0}".format(" ".join(print_dashes)))
```

Comment:
Why not rename `hangman_print_dashes` to `hangman_print_hint` and pass it `hint` to print instead of `word`? So for the case when the guess is wrong or already used, you can do `hangman_print_hint("_" * len(word))` and for the case where the guess is correct you can pass the updated hint. This way you also do not need `print_dashes_list` I think.

Reply:
Aha, I see you might have mixed up the meaning of hints versus dashes! The hint just tells the player what theme the word is taken from. It's printed for every round of the game! 

The print_dashes method prints the number of dashes "_" on the screen that gets updated based on the correct guesses. So they have different functionality and hence separate!